src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/AutoDisenchanter.java

```diff
@@ -73,8 +84,26 @@ protected MachineRecipe findNextRecipe(BlockMenu menu) {
                 int amount = 0;
 
                 for (Map.Entry<Enchantment, Integer> entry : item.getEnchantments().entrySet()) {
-                    enchantments.put(entry.getKey(), entry.getValue());
-                    amount++;
+                    if (!useEnchantLevelLimit.getValue() || enchantLevelLimit.getValue() >= entry.getValue()) {
+                        enchantments.put(entry.getKey(), entry.getValue());
+                        amount++;
+                    } else if (!menu.toInventory().getViewers().isEmpty()) {
+                        String notice = ChatColor.translateAlternateColorCodes('&', SlimefunPlugin.getLocalization().getMessage("messages.above-limit-level")
+                                .replace("%level%", enchantLevelLimit.getValue().toString()));
+
+                        ItemStack progressBar = getProgressBar().clone();
```

Comment:
Not really sure cloning is helpful here. You overwrite all the data anyway and this is more expensive than just making a new object.